modules/azure/actiongroup.go

```diff
@@ -0,0 +1,57 @@
+package azure
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/resources/mgmt/insights"
+	"github.com/stretchr/testify/require"
+)
+
+// GetActionGroupResource gets the AppServicePlan.
+// ruleName - required to find the AppServicePlan.
+// resGroupName - use an empty string if you have the AZURE_RES_GROUP_NAME environment variable set
+// subscriptionId - use an empty string if you have the ARM_SUBSCRIPTION_ID environment variable set
```

Comment:
Let me double check on this one. I know I had this scenario about a month ago, but I don't remember how it was handled. I would assume we want this to fail the test if it doesn't exist since that is the purpose of this function right?

Reply:
Either way is fine as long as we are consistent, and it should be documented in the comments.

The reason it's not a huge issue is that both paths are equally workable: E.g., if it fails the test, one could use the `E` version to check for non-existence. If it returns `nil`, one could assert it is nil to fail the test.

Reply:
Added unit tests to cover this. It throws an error when no Resource Group is found.